### PR TITLE
allow repository_dispatch to trigger work flow when gateway Config.go is changed

### PR DESCRIPTION
**Pull request by yurisasuke**
## Related Issue (if applicable)
https://tyktech.atlassian.net/browse/TT-8253

## Description
Currently, we have to trigger the config_gen workflow manually whenever the config.go changes in the tyk gateway. This change will allow us to receive repository_dispatch from tyk gateway that will trigger the workflow.

## Types of changes

- [ ] Fixing typo (please merge to production)
- [ ] Documenting a new feature (please merge to production)
- [ ] Documentation for future release (please do not merge to production)
- [ ] Something else (please add if needs merging to production or not)

## Checklist

- [ ] I have [reviewed the guidelines](../CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](../CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I have added a preview link.

### CODE REVIEW

File: .github/workflows/config_gen.yaml [hunk 1]
```diff
@@ -17,38 +19,48 @@ on:
...
+          echo "DOC_BRANCH=${{ github.event.inputs.branch }}" >> $GITHUB_ENV
 
-    - name: Checkout Config Generator Repo
-      uses: actions/checkout@master
-      with:
-        repository: TykTechnologies/tyk-config-info-generator
-        path: ./tyk-config-info-generator
-        token: ${{ secrets.ORG_GH_TOKEN }}
+      - name: Checkout
+        uses: actions/checkout@master
```
Comment (letzya) [inline .github/workflows/config_gen.yaml]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/config_gen.yaml [hunk 2]
```diff
@@ -17,38 +19,48 @@ on:
...
+          token: ${{ secrets.ORG_GH_TOKEN }}
+          ref: ${{ env.DOC_BRANCH }}
 
-    - name: Generate markdown
-      run: |
-        cd ./tyk-config-info-generator/src
-        sudo TOKEN=${{ secrets.ORG_GH_TOKEN }} node app.js ${{ github.event.inputs.repo }}:${{ github.event.inputs.repoBranch }}
-        cp /node/home/tyk-config-info-generator/info/${{ github.event.inputs.repoBranch }}/${{ github.event.inputs.repo }}.md ../../tyk-docs/tyk-docs/content/shared/${{ github.event.inputs.repo }}-config.md
+      - name: Checkout Config Generator Repo
+        uses: actions/checkout@master
```
Comment (letzya) [inline .github/workflows/config_gen.yaml]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/config_gen.yaml [hunk 3]
```diff
@@ -17,38 +19,48 @@ on:
...
-      BRANCH: ${{ github.event.inputs.repoBranch }}
     runs-on: ubuntu-latest
     steps:
-    - name: Checkout
-      uses: actions/checkout@master
-      with:
-        path: ./tyk-docs
-        token: ${{ secrets.ORG_GH_TOKEN }}
-        ref: ${{ github.event.inputs.branch }}
+      - name: Set Repository Dispatch ENV
```
Comment (letzya) [inline .github/workflows/config_gen.yaml:24]: @Keithwachira you chose to do this login in 2 separate steps. What do you think about one step using `else`?

File: .github/workflows/config_gen.yaml [hunk 4]
```diff
@@ -17,38 +19,48 @@ on:
...
+        run: |
+          cd ./tyk-config-info-generator/src
+          sudo TOKEN=${{ secrets.ORG_GH_TOKEN }} node app.js ${{ env.REPO }}:${{ env.BRANCH }}
+          cp /node/home/tyk-config-info-generator/info/${{ env.BRANCH }}/${{ env.REPO }}.md ../../tyk-docs/tyk-docs/content/shared/${{ env.REPO }}-config.md
+
+      - name: Raise configuration changes Pull Request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          commit-message: update config
+          title: Import config documentation from ${{ env.REPO  }}:${{ env.BRANCH  }}
```
Comment (letzya) [inline .github/workflows/config_gen.yaml]: Can we add to the title something to show if it was triggered automatically or manually (based on the even type)

File: .github/workflows/config_gen.yaml (hunk 3 above)
Comment (yurisasuke) [reply .github/workflows/config_gen.yaml:24]: Unfortunately GitHub actions  do not have if else : [Reference](https://stackoverflow.com/questions/60916931/github-action-does-the-if-have-an-else)

File: .github/workflows/config_gen.yaml (hunk 4 above)
Comment (yurisasuke) [reply .github/workflows/config_gen.yaml]: > Can we add to the title something to show if it was triggered automatically or manually (based on the even type)

Yes, this is possible let me add it.

**Review by letzya — APPROVED** [review]

**Pull request merged**